app/src/main/java/org/mozilla/rocket/content/ContentPortalView.kt

```diff
@@ -0,0 +1,199 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.rocket.content
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Context
+import android.support.design.widget.BottomSheetBehavior
+import android.support.design.widget.CoordinatorLayout
+import android.support.v4.app.FragmentActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.util.AttributeSet
+import android.view.View
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.widget.Toast
+import kotlinx.android.synthetic.main.content_portal.view.*
+import org.mozilla.focus.R
+import org.mozilla.focus.bookmark.BookmarkAdapter
+import org.mozilla.focus.fragment.PanelFragment
+import org.mozilla.focus.navigation.ScreenNavigator
+import org.mozilla.focus.persistence.BookmarkModel
+import org.mozilla.focus.persistence.BookmarksDatabase
+import org.mozilla.focus.repository.BookmarkRepository
+import org.mozilla.focus.viewmodel.BookmarkViewModel
+import org.mozilla.rocket.content.ContentPortalViewState.isLastSessionContent
+
+object ContentPortalViewState {
+    @JvmStatic
+    var isLastSessionContent = false
```

Comment:
My suggestion will be move this to an independent class, and have strings representing sources that enter browsing ("Launcher", "news-portal", "history", "screenshot-viewe")...etc